floor/floor/controller/midi/functions.py

```diff
@@ -26,64 +27,94 @@ def add(cls, name, **kwargs):
                   help_text='Go to playlist position 4.')
 MidiFunctions.add('playlist_goto_5',
+                  callback=lambda controller, _: controller.playlist.go_to(5),
                   help_text='Go to playlist position 5.')
 MidiFunctions.add('playlist_goto_6',
+                  callback=lambda controller, _: controller.playlist.go_to(6),
                   help_text='Go to playlist position 6.')
 MidiFunctions.add('playlist_goto_7',
+                  callback=lambda controller, _: controller.playlist.go_to(7),
                   help_text='Go to playlist position 7.')
 MidiFunctions.add('playlist_goto_8',
+                  callback=lambda controller, _: controller.playlist.go_to(8),
                   help_text='Go to playlist position 8.')
 MidiFunctions.add('playlist_goto_9',
+                  callback=lambda controller, _: controller.playlist.go_to(9),
                   help_text='Go to playlist position 9.')
 MidiFunctions.add('playlist_goto_10',
+                  callback=lambda controller, _: controller.playlist.go_to(10),
                   help_text='Go to playlist position 10.')
 MidiFunctions.add('playlist_goto_11',
+                  callback=lambda controller, _: controller.playlist.go_to(11),
                   help_text='Go to playlist position 11.')
 MidiFunctions.add('playlist_goto_12',
+                  callback=lambda controller, _: controller.playlist.go_to(12),
                   help_text='Go to playlist position 12.')
 MidiFunctions.add('playlist_goto_13',
+                  callback=lambda controller, _: controller.playlist.go_to(13),
                   help_text='Go to playlist position 13.')
 MidiFunctions.add('playlist_goto_14',
+                  callback=lambda controller, _: controller.playlist.go_to(14),
                   help_text='Go to playlist position 14.')
 MidiFunctions.add('playlist_goto_15',
+                  callback=lambda controller, _: controller.playlist.go_to(15),
                   help_text='Go to playlist position 15.')
 MidiFunctions.add('playlist_goto_16',
+                  callback=lambda controller, _: controller.playlist.go_to(16),
                   help_text='Go to playlist position 16.')
 MidiFunctions.add('set_bpm',
+                  callback=lambda controller, value: controller.set_bpm(90 + 80 * (value/127.0)),
                   help_text='Set the global bpm based on note velocity or controller value.')
 MidiFunctions.add('set_brightness',
+                  callback=lambda controller, value: controller.scale_brightness(value/127.0),
                   help_text='Adjust the max brightness of the floor')
 
+# Add a number of ranged values, likely faders or knobs, that can be used within processor code
+# to tweak parameters.
+for idx in range(0, 4):
+    MidiFunctions.add('ranged_value_{}'.format(idx+1),
+                      callback=lambda controller, value, i=idx: controller.handle_ranged_value(i, value),
```

Comment:
This was weird for me; In most languages if you use a value from current scope in a lambda it creates a closure for that variable.   However, when I used `idx` without passing it as a arg, it would use the last value `idx` had.  In this case because we re-use `idx` below in a `range(1, 65)`, the last value of `idx` was 64, and that is what gets passed in to handle_ranged_value.

Passing in as a parameter with this default essentially creates the closure I want and without the future caller having to worry about it.